notifique/src/main/java/com/nathanrassi/notifique/NotifiqueListenerService.kt

```diff
@@ -2,19 +2,68 @@ package com.nathanrassi.notifique
 
 import android.service.notification.NotificationListenerService
 import android.service.notification.StatusBarNotification
+import android.util.Log
 import dagger.android.AndroidInjection
 import javax.inject.Inject
 
 class NotifiqueListenerService : NotificationListenerService() {
   @Inject internal lateinit var dao: Notifique.Dao
+  val TAG: String = "Notifique"
+  val notificationMap = mutableMapOf<String,MutableList<Int>>()
 
   override fun onCreate() {
     AndroidInjection.inject(this)
   }
 
   override fun onNotificationPosted(sbn: StatusBarNotification) {
+    val packageName: String = sbn.packageName.toString()
+    val notificationID: Int = sbn.id
+    if (uniqueNotification(packageName, notificationID)) {
+      //create notification object for db call
+      val message: String? = sbn.notification.extras.getString(android.app.Notification.EXTRA_TEXT).toString()
+      val title: String? = sbn.notification.extras.getString(android.app.Notification.EXTRA_TITLE).toString()
+      if (!message.isNullOrEmpty() && !title.isNullOrEmpty()) {
+        val notification: Notifique = Notifique(message, title, packageName, sbn.postTime)
+        dao.insert(notification)
+      } else {
+        Log.e(TAG, "message or title null/empty, nothing to add")
+      }
+    }
+
   }
 
   override fun onNotificationRemoved(sbn: StatusBarNotification) {
+    val packageName: String = sbn.packageName.toString()
+    val notificationID: Int = sbn.id
+    removeNotification(packageName, notificationID)
+  }
+
+
+  private fun uniqueNotification(pkg: String, id: Int): Boolean{
+    if (notificationMap.containsKey(pkg)) {
```

Comment:
do Map.get first instead of the contains. if the value is null, it doesn't contain it. then, you don't have to do the extra assertions for nullability below.